ReactiveXComponent/Connection/XCPublisher.cs

```diff
@@ -1,61 +1,33 @@
-﻿using System;
-using System.IO;
-using ReactiveXComponent.Common;
+﻿using ReactiveXComponent.Common;
+using ReactiveXComponent.Parser;
 
 namespace ReactiveXComponent.Connection
 {
     public class XCPublisher : IXCPublisher
     {
-        private bool _disposed;
-        private string _privateCommunicationIdentifier;
-        //private readonly string _component;
-        //private readonly DeploymentParser _parser
+        public static string PrivateCommunicationIdentifier;
+        private readonly DeploymentParser _parser;
 
-        public XCPublisher(string component, Stream xcApiStream)
-        {
-            //_component = component;
-            //_parser = new DeploymentParser(xcApiStream);
-        }
+        public DeploymentParser Parser => _parser;
 
-        public void InitPrivateCommunication(string privateCommunicationIdentifier)
+        public XCPublisher(DeploymentParser parser)
         {
-            _privateCommunicationIdentifier = privateCommunicationIdentifier;
+            _parser = parser;
```

Comment:
instead of using a parser everywhere, you should create an XCConfiguration class (that uses your parser). publishers and subscribers just need to know the configuration (check js api)